wwwapp/models.py

```diff
@@ -281,3 +301,28 @@ def result_in_percent(self):
 
     class Meta:
         unique_together = [('workshop', 'participant')]
+
+
+class ResourceYearPermission(models.Model):
+    """
+    Resource asociated with a WWW edition (year). Resource can be accessed by
+    users who qualified or had a lecture on year.
+    """
+    display_name = models.CharField(max_length=50, blank=True)
+    access_url = models.URLField(blank=True,
+                                 help_text="URL dla przycisku w menu. Przycisk nie jest wyświetlany jeśli url jest pusty")
+    root_url = models.CharField(max_length=256, null=False, blank=False,
```

Comment:
I don't like this name (this is not an URL, only a part of it), and isn't access_url basically root_url + hostname?

Reply:
root_url contains also hostname and port. (like warsztatywww.pl/asdasd) `access_url` is a proper URL for link. Maybe I could add an example to the comment.

Reply:
Oh, I didn't notice you changed it (that's why this needed better docs :P). Why not a full URL then, it's confusing to have a hostname without http://

Are you even checking it anywhere? X-Original-URI doesn't have the hostname in default configuration iirc

In that case, maybe public_url and private_url or access_url and internal_url (the second one sounds better imho)?

Reply:
So what does X-Original-URI actually contain? At first I implemented for a case where it's only a path, now i implemented for URL.

Reply:
Well... like I said a few times already, it's the $request_uri variable from nginx which is the request URI part of the request as defined in [RFC 2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html#sec5.1.2) :P

Tl;dr it's `/internet/www14/something?query=string` (the path and query string parts of the URL)

Reply:
So it was implemented properly before...

Reply:
It wasn't - you were trying to interpret this URI a a file path in the filesystem with INTERNET_MNT_PATH. The code path where this variable was None was correct though.

Reply:
Or actually, it wasn't either, it would break with query strings too. The current implementation is better if you just get rid of the hostname. It may be a good idea to make root_url always start with a `/` to make it look more like an actual request URI, although that's not required from functionality point of view.